res/controllers/Numark-Mixtrack-Pro-scripts.js

```diff
@@ -54,655 +60,1120 @@
-
-//	print ("Delay: " + engine.getValue("[Flanger]","lfoDelay"));
-
-		if (engine.getValue("[Flanger]","lfoDelay") < 5026) {
-
-			speed = engine.getValue("[Flanger]","lfoDelay") / 5025;
-
-			if (speed < 0) speed = 0;
-
-		} else {
-
-			speed = (engine.getValue("[Flanger]","lfoDelay") - 5009)/ 16,586666667
-
-			if (speed > 300) speed = 300;
-		}
-
-//	print ("Speed: " + speed);
-
-		engine.brake(deck, value, speed);
-
-		if (!value) NumarkMixTrackPro.toggleDeleteKey(channel, control, 1, status, group);
-
-	} else {
-		if (!value) return;
-		if (engine.getValue(group, "flanger")) {
-			engine.setValue(group, "flanger", 0);
-		}else{
-			engine.setValue(group, "flanger", 1);
-		}
-	}
-
-}
-
-
-NumarkMixTrackPro.cuebutton = function(channel, control, value, status, group) {
-
-
-	// Don't set Cue accidentally at the end of the song
-	if (engine.getValue(group, "playposition") <= 0.97) {
-			engine.setValue(group, "cue_default", value ? 1 : 0);
-	} else {
-		engine.setValue(group, "cue_preview", value ? 1 : 0);
-	}
-
-}
+    status = status ? 0x64 : 0x00;
+    midi.sendShortMsg(0x90, value, status);
+};
+
+NumarkMixTrackPro.flashLED = function(led, veces) {
+    const ndx = Math.random();
+    const id = engine.beginTimer(120, NumarkMixTrackPro.doFlash(ndx, veces));
+    NumarkMixTrackPro.ledTimers[ndx] = new NumarkMixTrackPro.LedTimer(
+        id,
+        led,
+        0,
+        false
+    );
+};
+
+NumarkMixTrackPro.doFlash = function(ndx, veces) {
+    const ledTimer = NumarkMixTrackPro.ledTimers[ndx];
+
+    if (!ledTimer) {
+        return;
+    }
+
+    if (ledTimer.count > veces) {
+        // how many times blink the button
+        engine.stopTimer(ledTimer.id);
+        delete NumarkMixTrackPro.ledTimers[ndx];
+    } else {
+        ledTimer.count++;
+        ledTimer.state = !ledTimer.state;
+        NumarkMixTrackPro.setLED(ledTimer.led, ledTimer.state);
+    }
+};
+
+NumarkMixTrackPro.selectKnob = function(
+    channel,
+    control,
+    value,
+    // status,
+    // group
+) {
+    if (value > 63) {
+        value = value - 128;
+    }
+    if (NumarkMixTrackPro.directoryMode) {
+        // [Playlist]SelectNextPlaylist and SelectPrevPlaylist are deprecaded
+        // but it respect LED on controller, [Library]MoveDown and MoveUp don't.
+        if (value > 0) {
+            for (let i = 0; i < value; i++) {
+                script.triggerControl("[Playlist]", "SelectNextPlaylist", 50);
+            }
+        } else {
+            for (let i = 0; i < -value; i++) {
+                script.triggerControl("[Playlist]", "SelectPrevPlaylist", 50);
+            }
+        }
+    } else {
+        engine.setParameter("[Playlist]", "SelectTrackKnob", value);
+    }
+};
+
+NumarkMixTrackPro.LoadTrack = function(
+    channel,
+    control,
+    value,
+    status,
+    group
+) {
+    // Load the selected track in the corresponding deck only if the track is paused
+
+    if (value && engine.getParameter(group, "play") !== 1) {
```

Comment:
we already have a setting for that. 
```suggestion
    if (value) {
```